### PR TITLE
Changed fish launching mechanism

### PR DESCRIPTION
**Pull request by sweetgiorni**
Before, fishing loot wouldn't fly towards the player fast enough; it would appear next to the floater and then sink. I added some magic numbers and now the fish fly at a much more appropriate velocity. 

There was also the problem that depending on how long the player waited before reeling in the loot, the floater could be deep underwater or closer to the surface. This resulted in the loot having inconsistent velocities due to the extra drag the water adds. I resolved this by saving the position of the floater just before it sinks underwater and spawning the loot there instead.
 :cloud: :fish:  :cloud:

### CODE REVIEW

File: Server/Plugins/APIDump/APIDesc.lua [hunk 1]
```diff
@@ -6585,6 +6585,15 @@ cFile:DeleteFile("/usr/bin/virus.exe");
...
 					Notes = "Returns the EntityID of the player who owns the floater.",
 				},
+				GetBitePos =
+				{
+					Returns =
+					{
+							Name = "BitePosition",
+							Type = "Vector3d",
+					},
+					Notes = "Returns the position of the floater just before a fish bites.",
```
Comment (NiLSPACE) [inline Server/Plugins/APIDump/APIDesc.lua]: Perhaps also provide info what it returns if a fish hasn't bitten yet?

File: src/Items/ItemFishingRod.h [hunk 2]
```diff
@@ -236,9 +239,11 @@ class cItemFishingRodHandler :
 				{
 					return true;
 				}
-				Vector3d FloaterPos = FloaterInfo.GetPos();
+				Vector3d FloaterPos = FloaterInfo.GetBitePos();
+				const float FISH_SPEED_MULT_XZ = 2.7f;
+				const float FISH_SPEED_MULT_Y = 2.0f;
```
Comment (NiLSPACE) [inline src/Items/ItemFishingRod.h]: Do these numbers work correctly even while the player is closer or further away from the floater?

File: Server/Plugins/APIDump/APIDesc.lua (hunk 1 above)
Comment (sweetgiorni) [inline Server/Plugins/APIDump/APIDesc.lua]: Fixed

File: src/Items/ItemFishingRod.h (hunk 2 above)
Comment (sweetgiorni) [inline src/Items/ItemFishingRod.h]: I tweaked it again a little bit but yes, for the most part it scales well for all _reasonable_ heights and distances. Although, the minecraft wiki says that moving 32 blocks away should break the line, but that doesn't happen. That's probably the next thing I'll work on.

**Comment by sphinxc0re** [comment]
@sweetgiorni any updates on this one?

**Pull request merged**